client/my-sites/stats/jetpack-upsell-section/index.tsx

```diff
@@ -27,11 +28,20 @@ export default function JetpackUpsellSection() {
 	const siteSlug = useSelector( getSelectedSiteSlug );
 
 	// NOTE: This will only work within Odyssey Stats.
+	// Doesn't recogize Commercial plan bundles.
 	const { purchasedProducts } = usePurchasedProducts();
 
+	// Check for bundled products.
+	// Currently checking for variations of the Complete bundle only.
+	const siteId = useSelector( getSelectedSiteId );
+	const sitePurchases = useSelector( ( state ) => getSitePurchases( state, siteId ) );
```

Comment:
It would be great to consolidate this purchase check into `useStatsPurchases` for plan support visibility and reusability. What do you think?

Reply:
I'll look at this today. I see that this hook uses the `filterPurchasesByProducts` function noted in your previous comment so two comments with one stone. ;>

Reply:
So I've looked at extending the `useStatsPurchases` hook for this but I'm not sure it makes sense here. The naming of the hook makes it unclear that it would provide data on other plans and it is obscuring the actual plan data which is what I'd prefer to work with.

I see two approaches that might make sense:

1. Adding additional flags for `hasCompletePlan`, `hasProfessionalPlan`, `hasSecurityBundle`, etc. as needed OR…
2. Adding a `data` value and returning the underlying results from `getSitePurchases` along with the various flags.

I'd lean towards option 2 but my real preference would be to call `getSitePurchases` and work directly with the returned data.

Is there a reason to prefer extending `useStatsPurchases` over calling the `getSitePurchases` directly?

Reply:
We don't need to use the hook `useStatsPurchases` here, but importing an independent check function seems an option. 🤔